robotic_hand:  
  # Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50

  # Position Controllers ---------------------------------------
  joint1_position_controller:
    type: effort_controllers/JointPositionController
    joint: base_link__link_01
    pid: {p: 1300, i: 500, d: 200}
  joint2_position_controller:
    type: effort_controllers/JointPositionController
    joint: link_01__link_02
    pid: {p: 2000, i: 70, d: 300}
  joint3_position_controller:
    type: effort_controllers/JointPositionController
    joint: link_02__link_03
    pid: {p: 1200, i: 13, d: 23}
  joint4_position_controller:
    type: effort_controllers/JointPositionController
    joint: link_03__link_04
    pid: {p: 6.7, i: 5.3, d: 2.0}